backend/src/libs/types/repository.type.ts

```diff
@@ -1,5 +1,5 @@
 type Repository<T = unknown> = {
-  find(): Promise<T>;
+  find(id: number): Promise<T>;
```

Comment:
What if we want to find a user by email? (Or any other field)

Reply:
in this case, should I create a separate method for finding users by id or change this method to be more generic?

Reply:
You can create a method findById, as @v0ldemar01 mentioned in one of his comments